CONTRIBUTING.md

```diff
@@ -42,29 +42,76 @@ Maintainers can revert your changes if they feel they are not worth maintaining
 
 As mentioned before, you are expected to follow these specifications in order to make everyone's lives easier, it will also save you and us time, with having to make the changes and us having to tell you what to change. Thank you for reading this section.
 
-* As BYOND's Dream Maker is an object oriented language, code must be object oriented when possible in order to be more flexible when adding content to it. If you are unfamiliar with this concept, it is highly recommended you look it up.
-
-* You must write BYOND code with absolute pathing, like so:
+###Object Oriented code 
+As BYOND's Dream Maker is an object oriented language, code must be object oriented when possible in order to be more flexible when adding content to it. If you are unfamiliar with this concept, it is highly recommended you look it up.
+
+###All Byond paths must contain the full path.
+(ie: absolute pathing)
+
+Byond will allow you nest almost any type keyword into a block, such as:
+
+```c++
+datum
+	datum1
+		var
+			varname1 = 1
+			varname2
+			static
+				varname3
+				varname4
+		proc
+			proc1()
+				code
+			proc2()
+				code
+		
+		datum2
+			varname1 = 0
+			proc
+				proc3()
+					code
+			proc2()
+				..()
+				code
+```
 
-```C++
+The use of this is not allowed in this project as it makes finding definitions via full text searching next to impossible. The only exception is the variables of an object may be nested to the object, but must not nest further.
+
+The previous code made compliant:
+
+```c++
+/datum/datum1
+		var/varname1
+		var/varname2
+		var/static/varname3
+		var/static/varname4
+
+/datum/datum1/proc/proc1()
+	code
+/datum/datum1/proc/proc2()
+	code
+/datum/datum1/datum2
+	varname1 = 0
+/datum/datum1/datum2/proc/proc3()
+	code
+/datum/datum1/datum2/proc2()
+	..()
+	code
+```
 
-/obj/item/weapon/baseball_bat
-    name = "baseball bat"
-    desc = "A baseball bat."
-    var/wooden = 1
+###No overriding byond type safety checks.
+The use of the : operator to override type safety checks is strongly discouraged. You must cast the variable to the proper type.
 
-/obj/item/weapon/baseball_bat/examine()
-    if(wooden)
-        desc = "A wooden baseball bat."
-    else
-        desc = "A metal baseball bat."
-    ..()
+Exceptions are only made when used in loops that require the performance boast from being called ***extremely*** often. (Rule of thumb: If you aren't messing with the master controller or it's subsystems, this exception probably doesn't apply)
```

Comment:
Capitalize BYOND, or use DM.  Boost, not boast.  >implying there is actually any form of type safety in DM